File: suc/kotlin/031.kt
package com.example.singlefileui

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

private const val NAME = "032_WalletDashFinance_en"
private const val UI_TYPE = "Finance/Wallet"
private const val STYLE_THEME = "Monochrome Comfortable"
private const val LANG = "en"
private const val BASELINE_SIZE = "720x1280"

object AppTokens {
    object Colors {
        val primary = Color(0xFF020617)
        val secondary = Color(0xFF4ADE80)
        val tertiary = Color(0xFF64748B)
        val background = Color(0xFFF8FAFC)
        val surface = Color(0xFFFFFFFF)
        val surfaceVariant = Color(0xFFF1F5F9)
        val outline = Color(0xFFE2E8F0)
        val success = Color(0xFF22C55E)
        val warning = Color(0xFFF59E0B)
        val error = Color(0xFFEF4444)
        val onPrimary = Color(0xFFFFFFFF)
        val onSecondary = Color(0xFF020617)
        val onTertiary = Color(0xFFFFFFFF)
        val onBackground = Color(0xFF020617)
        val onSurface = Color(0xFF020617)
    }

    object TypographyTokens {
        val display = TextStyle(fontSize = 48.sp, fontWeight = FontWeight.Bold, lineHeight = 56.sp, letterSpacing = (-0.5).sp)
        val headline = TextStyle(fontSize = 28.sp, fontWeight = FontWeight.SemiBold, lineHeight = 36.sp)
        val title = TextStyle(fontSize = 20.sp, fontWeight = FontWeight.Medium, lineHeight = 28.sp)
        val body = TextStyle(fontSize = 16.sp, fontWeight = FontWeight.Normal, lineHeight = 24.sp)
        val label = TextStyle(fontSize = 12.sp, fontWeight = FontWeight.Medium, lineHeight = 16.sp, letterSpacing = 0.2.sp)
    }

    object Shapes {
        val small = RoundedCornerShape(8.dp)
        val medium = RoundedCornerShape(16.dp)
        val large = RoundedCornerShape(24.dp)
    }

    object Spacing {
        val xs = 4.dp
        val sm = 8.dp
        val md = 16.dp
        val lg = 24.dp
        val xl = 32.dp
        val xxl = 48.dp
    }

    data class ShadowSpec(val elevation: Dp, val radius: Dp, val dy: Dp, val opacity: Float)
    object ElevationMapping {
        val level1 = ShadowSpec(1.dp, 3.dp, 1.dp, 0.05f)
        val level2 = ShadowSpec(4.dp, 8.dp, 2.dp, 0.08f)
        val level3 = ShadowSpec(8.dp, 12.dp, 4.dp, 0.10f)
    }
}

private val AppColorScheme = lightColorScheme(
    primary = AppTokens.Colors.primary,
    onPrimary = AppTokens.Colors.onPrimary,
    secondary = AppTokens.Colors.secondary,
    onSecondary = AppTokens.Colors.onSecondary,
    tertiary = AppTokens.Colors.tertiary,
    onTertiary = AppTokens.Colors.onTertiary,
    background = AppTokens.Colors.background,
    onBackground = AppTokens.Colors.onBackground,
    surface = AppTokens.Colors.surface,
    onSurface = AppTokens.Colors.onSurface,
    surfaceVariant = AppTokens.Colors.surfaceVariant,
    outline = AppTokens.Colors.outline,
    error = AppTokens.Colors.error
)

private val AppTypography = Typography(
    displayLarge = AppTokens.TypographyTokens.display,
    headlineMedium = AppTokens.TypographyTokens.headline,
    titleMedium = AppTokens.TypographyTokens.title,
    bodyMedium = AppTokens.TypographyTokens.body,
    labelMedium = AppTokens.TypographyTokens.label
)

@Composable
fun AppTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = AppColorScheme,
        typography = AppTypography,
        shapes = Shapes(
            small = AppTokens.Shapes.small,
            medium = AppTokens.Shapes.medium,
            large = AppTokens.Shapes.large
        ),
        content = content
    )
}

data class Transaction(val id: Int, val name: String, val detail: String, val amount: String, val isCredit: Boolean)
data class ControlAction(val label: String)
data class NavItem(val label: String)

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun RootScreen() {
    val transactions = remember {
        listOf(
            Transaction(1, "Spotify", "Subscription", "-$9.99", false),
            Transaction(2, "Income", "Monthly Salary", "+$2,500.00", true),
            Transaction(3, "Starbucks", "Coffee", "-$5.75", false),
            Transaction(4, "Amazon", "Shopping", "-$124.50", false),
            Transaction(5, "Refund", "Amazon Return", "+$32.00", true)
        )
    }
    val controlActions = remember { listOf(ControlAction("Send"), ControlAction("Receive"), ControlAction("Add"), ControlAction("More")) }
    val navItems = remember { listOf(NavItem("Home"), NavItem("Cards"), NavItem("Activity"), NavItem("Profile")) }
    val selectedNavItem = remember { mutableStateOf("Home") }

    Scaffold(
        contentWindowInsets = WindowInsets(0),
        containerColor = AppTokens.Colors.background,
        topBar = {
            CenterAlignedTopAppBar(
                title = { Text("My Wallet", style = MaterialTheme.typography.titleMedium) },
                navigationIcon = {
                    IconButton(onClick = {}) {
                        Box(modifier = Modifier.size(28.dp).background(AppTokens.Colors.surfaceVariant, CircleShape))
                    }
                },
                actions = {
                    IconButton(onClick = {}) {
                        Box(modifier = Modifier.size(28.dp).background(AppTokens.Colors.surfaceVariant, CircleShape))
                    }
                },
                colors = TopAppBarDefaults.centerAlignedTopAppBarColors(containerColor = Color.Transparent)
            )
        },
        bottomBar = {
            NavigationBar(
                containerColor = AppTokens.Colors.surface,
                tonalElevation = AppTokens.ElevationMapping.level3.elevation,
            ) {
                navItems.forEach { item ->
                    NavigationBarItem(
                        selected = selectedNavItem.value == item.label,
                        onClick = { selectedNavItem.value = item.label },
                        icon = {
                            Box(modifier = Modifier.size(24.dp).background(
                                if(selectedNavItem.value == item.label) AppTokens.Colors.primary else AppTokens.Colors.outline,
                                CircleShape)
                            )
                        },
                        label = { Text(item.label, style = MaterialTheme.typography.labelMedium) },
                        colors = NavigationBarItemDefaults.colors(
                            selectedIconColor = AppTokens.Colors.primary,
                            unselectedIconColor = AppTokens.Colors.tertiary,
                            selectedTextColor = AppTokens.Colors.primary,
                            unselectedTextColor = AppTokens.Colors.tertiary,
                            indicatorColor = AppTokens.Colors.surfaceVariant
                        )
                    )
                }
            }
        }
    ) { paddingValues ->
        LazyColumn(
            modifier = Modifier.fillMaxSize().padding(paddingValues),
            contentPadding = PaddingValues(horizontal = AppTokens.Spacing.md, vertical = AppTokens.Spacing.sm),
            verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.lg)
        ) {
            item {
                Card(
                    modifier = Modifier.fillMaxWidth(),
                    shape = AppTokens.Shapes.large,
                    colors = CardDefaults.cardColors(containerColor = AppTokens.Colors.primary),
                    elevation = CardDefaults.cardElevation(defaultElevation = AppTokens.ElevationMapping.level2.elevation)
                ) {
                    Column(modifier = Modifier.padding(AppTokens.Spacing.lg)) {
                        Text(
                            "Total Balance",
                            style = MaterialTheme.typography.bodyMedium,
                            color = AppTokens.Colors.onPrimary.copy(alpha = 0.7f)
                        )
                        Spacer(modifier = Modifier.height(AppTokens.Spacing.sm))
                        Text(
                            "$ 1,145.14",
                            style = MaterialTheme.typography.displayLarge,
                            color = AppTokens.Colors.onPrimary
                        )
                        Spacer(modifier = Modifier.height(AppTokens.Spacing.xs))
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Box(modifier = Modifier.size(10.dp).background(AppTokens.Colors.secondary, CircleShape))
                            Spacer(modifier = Modifier.width(AppTokens.Spacing.sm))
                            Text("+$234.10 today", style = MaterialTheme.typography.bodyMedium, color = AppTokens.Colors.secondary)
                        }
                    }
                }
            }

            item {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceAround,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    controlActions.forEach { action ->
                        Column(
                            horizontalAlignment = Alignment.CenterHorizontally,
                            verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm)
                        ) {
                            Button(
                                onClick = {},
                                modifier = Modifier.size(64.dp),
                                shape = CircleShape,
                                colors = ButtonDefaults.buttonColors(containerColor = AppTokens.Colors.surface),
                                elevation = ButtonDefaults.buttonElevation(defaultElevation = AppTokens.ElevationMapping.level1.elevation),
                                contentPadding = PaddingValues(0.dp)
                            ) {
                                Box(modifier = Modifier.size(24.dp).background(AppTokens.Colors.primary, AppTokens.Shapes.small))
                            }
                            Text(action.label, style = MaterialTheme.typography.labelMedium)
                        }
                    }
                }
            }

            item {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text("Recent Activity", style = MaterialTheme.typography.titleMedium)
                    TextButton(onClick = {}) {
                        Text("View All", style = MaterialTheme.typography.labelMedium, color = AppTokens.Colors.tertiary)
                    }
                }
            }

            items(transactions) { transaction ->
                Card(
                    modifier = Modifier.fillMaxWidth(),
                    shape = AppTokens.Shapes.medium,
                    colors = CardDefaults.cardColors(containerColor = AppTokens.Colors.surface),
                    border = BorderStroke(1.dp, AppTokens.Colors.outline)
                ) {
                    Row(
                        modifier = Modifier.padding(AppTokens.Spacing.md),
                        verticalAlignment = Alignment.CenterVertically,
                        horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.md)
                    ) {
                        Box(modifier = Modifier.size(48.dp).background(AppTokens.Colors.surfaceVariant, AppTokens.Shapes.medium))
                        Column(modifier = Modifier.weight(1f)) {
                            Text(transaction.name, style = MaterialTheme.typography.bodyMedium, fontWeight = FontWeight.SemiBold)
                            Text(transaction.detail, style = MaterialTheme.typography.labelMedium, color = AppTokens.Colors.tertiary)
                        }
                        Text(
                            transaction.amount,
                            style = MaterialTheme.typography.bodyMedium,
                            color = if (transaction.isCredit) AppTokens.Colors.success else AppTokens.Colors.onSurface,
                            fontWeight = FontWeight.SemiBold,
                            textAlign = TextAlign.End
                        )
                    }
                }
            }
        }
    }
}

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        WindowCompat.setDecorFitsSystemWindows(window, false)
        val controller = WindowInsetsControllerCompat(window, window.decorView)
        controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        controller.hide(WindowInsetsCompat.Type.systemBars())
        setContent {
            AppTheme {
                Surface(color = MaterialTheme.colorScheme.background) {
                    RootScreen()
                }
            }
        }
    }

    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus) {
            val controller = WindowInsetsControllerCompat(window, window.decorView)
            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
            controller.hide(WindowInsetsCompat.Type.systemBars())
        }
    }
}

@Preview(showBackground = true, widthDp = 360, heightDp = 720)
@Composable
fun PreviewRoot() {
    AppTheme {
        RootScreen()
    }
}

File: suc/swift/031.swift
import SwiftUI

// MARK: - AppTokens (设计系统令牌)
// 统一管理颜色、字体、形状、间距和阴影，便于全局修改和保持一致性。
struct AppTokens {
    struct Colors {
        static let primary = Color(red: 0x02 / 255.0, green: 0x06 / 255.0, blue: 0x17 / 255.0)
        static let secondary = Color(red: 0x4A / 255.0, green: 0xDE / 255.0, blue: 0x80 / 255.0)
        static let tertiary = Color(red: 0x64 / 255.0, green: 0x74 / 255.0, blue: 0x8B / 255.0)
        static let background = Color(red: 0xF8 / 255.0, green: 0xFA / 255.0, blue: 0xFC / 255.0)
        static let surface = Color(red: 0xFF / 255.0, green: 0xFF / 255.0, blue: 0xFF / 255.0)
        static let surfaceVariant = Color(red: 0xF1 / 255.0, green: 0xF5 / 255.0, blue: 0xF9 / 255.0)
        static let outline = Color(red: 0xE2 / 255.0, green: 0xE8 / 255.0, blue: 0xF0 / 255.0)
        static let success = Color(red: 0x22 / 255.0, green: 0xC5 / 255.0, blue: 0x5E / 255.0)
        static let warning = Color(red: 0xF5 / 255.0, green: 0x9E / 255.0, blue: 0x0B / 255.0)
        static let error = Color(red: 0xEF / 255.0, green: 0x44 / 255.0, blue: 0x44 / 255.0)
        static let onPrimary = Color(red: 0xFF / 255.0, green: 0xFF / 255.0, blue: 0xFF / 255.0)
        static let onSecondary = Color(red: 0x02 / 255.0, green: 0x06 / 255.0, blue: 0x17 / 255.0)
        static let onTertiary = Color(red: 0xFF / 255.0, green: 0xFF / 255.0, blue: 0xFF / 255.0)
        static let onBackground = Color(red: 0x02 / 255.0, green: 0x06 / 255.0, blue: 0x17 / 255.0)
        static let onSurface = Color(red: 0x02 / 255.0, green: 0x06 / 255.0, blue: 0x17 / 255.0)
    }

    struct TypographyTokens {
        // Compose的 lineHeight 在 SwiftUI 中没有直接对应属性。
        // 对于单行文本，font 本身已包含行高信息。
        // letterSpacing 对应 SwiftUI 的 kerning。
        static let display = Font.system(size: 48, weight: .bold) // lineHeight = 56.sp, letterSpacing = -0.5.sp
        static let headline = Font.system(size: 28, weight: .semibold) // lineHeight = 36.sp
        static let title = Font.system(size: 20, weight: .medium) // lineHeight = 28.sp
        static let body = Font.system(size: 16, weight: .regular) // lineHeight = 24.sp
        static let label = Font.system(size: 12, weight: .medium) // lineHeight = 16.sp, letterSpacing = 0.2.sp
    }

    struct Shapes {
        // 自定义 Shape 结构体，以便像 Compose 一样直接使用 `AppTokens.Shapes.small`
        struct RoundedCornerShape: Shape {
            let cornerRadius: CGFloat
            func path(in rect: CGRect) -> Path {
                Path(roundedRect: rect, cornerRadius: cornerRadius)
            }
        }
        static let small = RoundedCornerShape(cornerRadius: 8)
        static let medium = RoundedCornerShape(cornerRadius: 16)
        static let large = RoundedCornerShape(cornerRadius: 24)
    }

    struct Spacing {
        static let xs: CGFloat = 4
        static let sm: CGFloat = 8
        static let md: CGFloat = 16
        static let lg: CGFloat = 24
        static let xl: CGFloat = 32
        static let xxl: CGFloat = 48
    }

    struct ShadowSpec {
        let elevation: CGFloat // Compose 中的概念层级，SwiftUI 中主要通过 radius, x, y, opacity 实现
        let radius: CGFloat
        let dy: CGFloat // y 轴偏移
        let opacity: Double
    }

    struct ElevationMapping {
        static let level1 = ShadowSpec(elevation: 1, radius: 3, dy: 1, opacity: 0.05)
        static let level2 = ShadowSpec(elevation: 4, radius: 8, dy: 2, opacity: 0.08)
        static let level3 = ShadowSpec(elevation: 8, radius: 12, dy: 4, opacity: 0.10)
    }
}

// MARK: - Data Models (数据模型)
// 转换为 Swift struct，并遵循 Identifiable 协议以便在 ForEach 中使用。
struct Transaction: Identifiable {
    let id: Int
    let name: String
    let detail: String
    let amount: String
    let isCredit: Bool
}

struct ControlAction: Identifiable {
    let id = UUID() // 使用 UUID 作为唯一标识符
    let label: String
}

struct NavItem: Identifiable {
    let id = UUID() // 使用 UUID 作为唯一标识符
    let label: String
}

// MARK: - RootScreen View (主屏幕视图)
struct RootScreen: View {
    @State private var selectedNavItem: String = "Home" // 对应 Compose 的 remember { mutableStateOf("Home") }

    // 静态数据，对应 Compose 的 remember { listOf(...) }
    let transactions = [
        Transaction(id: 1, name: "Spotify", detail: "Subscription", amount: "-$9.99", isCredit: false),
        Transaction(id: 2, name: "Income", detail: "Monthly Salary", amount: "+$2,500.00", isCredit: true),
        Transaction(id: 3, name: "Starbucks", detail: "Coffee", amount: "-$5.75", isCredit: false),
        Transaction(id: 4, name: "Amazon", detail: "Shopping", amount: "-$124.50", isCredit: false),
        Transaction(id: 5, name: "Refund", detail: "Amazon Return", amount: "+$32.00", isCredit: true)
    ]
    let controlActions = [ControlAction(label: "Send"), ControlAction(label: "Receive"), ControlAction(label: "Add"), ControlAction(label: "More")]
    let navItems = [NavItem(label: "Home"), NavItem(label: "Cards"), NavItem(label: "Activity"), NavItem(label: "Profile")]

    var body: some View {
        ZStack {
            // Scaffold 的 containerColor 和 contentWindowInsets 0 对应
            AppTokens.Colors.background.ignoresSafeArea(.all)

            VStack(spacing: 0) {
                // 顶部栏 (TopAppBar)
                topAppBar()
                    .padding(.horizontal, AppTokens.Spacing.md) // 内部内容水平内边距
                    .padding(.vertical, AppTokens.Spacing.sm) // 内部内容垂直内边距
                    .background(AppTokens.Colors.background.opacity(0.01)) // 透明背景
                    .ignoresSafeArea(.container, edges: .top) // 延伸到安全区域顶部

                // 主要内容区域 (LazyColumn)
                ScrollView(.vertical, showsIndicators: false) {
                    VStack(spacing: AppTokens.Spacing.lg) { // verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.lg)
                        // 总余额卡片
                        totalBalanceCard()

                        // 控制操作按钮行
                        controlActionsRow()

                        // 最近活动标题
                        recentActivityHeader()

                        // 交易列表
                        ForEach(transactions) { transaction in
                            transactionCard(transaction: transaction)
                        }
                    }
                    .padding(.horizontal, AppTokens.Spacing.md) // contentPadding horizontal
                    .padding(.vertical, AppTokens.Spacing.sm) // contentPadding vertical
                }
                // 在 ScrollView 底部和导航栏之间添加一些间距，以匹配 Compose 的视觉效果
                .padding(.bottom, AppTokens.Spacing.sm)

                // 底部导航栏 (NavigationBar)
                bottomNavigationBar()
                    // tonalElevation 对应阴影效果
                    .shadow(color: .black.opacity(AppTokens.ElevationMapping.level3.opacity),
                            radius: AppTokens.ElevationMapping.level3.radius,
                            x: 0,
                            y: AppTokens.ElevationMapping.level3.dy)
                    .ignoresSafeArea(.container, edges: .bottom) // 延伸到安全区域底部
            }
        }
        // 全屏显示并隐藏顶部状态栏 (iOS 16.0 可用)
        .statusBarHidden(true)
    }

    // MARK: - Subviews (子视图)
    // 顶部应用栏
    private func topAppBar() -> some View {
        HStack {
            Button(action: {}) {
                Circle()
                    .fill(AppTokens.Colors.surfaceVariant)
                    .frame(width: 28, height: 28)
            }
            Spacer()
            Text("My Wallet")
                .font(AppTokens.TypographyTokens.title) // MaterialTheme.typography.titleMedium
                .foregroundColor(AppTokens.Colors.onBackground)
            Spacer()
            Button(action: {}) {
                Circle()
                    .fill(AppTokens.Colors.surfaceVariant)
                    .frame(width: 28, height: 28)
            }
        }
        .frame(height: 56) // 标准 TopAppBar 高度
    }

    // 总余额卡片
    private func totalBalanceCard() -> some View {
        VStack(alignment: .leading, spacing: 0) {
            Text("Total Balance")
                .font(AppTokens.TypographyTokens.body) // MaterialTheme.typography.bodyMedium
                .foregroundColor(AppTokens.Colors.onPrimary.opacity(0.7))
            Spacer().frame(height: AppTokens.Spacing.sm)
            Text("$ 1,145.14")
                .font(AppTokens.TypographyTokens.display) // MaterialTheme.typography.displayLarge
                .kerning(-0.5) // letterSpacing
                .foregroundColor(AppTokens.Colors.onPrimary)
            Spacer().frame(height: AppTokens.Spacing.xs)
            HStack(alignment: .center, spacing: AppTokens.Spacing.sm) {
                Circle()
                    .fill(AppTokens.Colors.secondary)
                    .frame(width: 10, height: 10)
                Text("+$234.10 today")
                    .font(AppTokens.TypographyTokens.body) // MaterialTheme.typography.bodyMedium
                    .foregroundColor(AppTokens.Colors.secondary)
            }
        }
        .padding(AppTokens.Spacing.lg)
        .frame(maxWidth: .infinity) // fillMaxWidth()
        .background(AppTokens.Colors.primary)
        .clipShape(AppTokens.Shapes.large) // shape = AppTokens.Shapes.large
        // elevation = CardDefaults.cardElevation(defaultElevation = AppTokens.ElevationMapping.level2.elevation)
        .shadow(color: .black.opacity(AppTokens.ElevationMapping.level2.opacity),
                radius: AppTokens.ElevationMapping.level2.radius,
                x: 0,
                y: AppTokens.ElevationMapping.level2.dy)
    }

    // 控制操作按钮行
    private func controlActionsRow() -> some View {
        HStack(alignment: .center, spacing: 0) { // horizontalArrangement = Arrangement.SpaceAround
            ForEach(controlActions) { action in
                VStack(spacing: AppTokens.Spacing.sm) {
                    Button(action: {}) {
                        // 按钮内部的 Box
                        AppTokens.Shapes.small
                            .fill(AppTokens.Colors.primary)
                            .frame(width: 24, height: 24)
                            .contentShape(Circle()) // 确保点击区域是圆形
                    }
                    .frame(width: 64, height: 64) // modifier = Modifier.size(64.dp)
                    .background(AppTokens.Colors.surface) // containerColor
                    .clipShape(Circle()) // shape = CircleShape
                    // elevation = ButtonDefaults.buttonElevation(defaultElevation = AppTokens.ElevationMapping.level1.elevation)
                    .shadow(color: .black.opacity(AppTokens.ElevationMapping.level1.opacity),
                            radius: AppTokens.ElevationMapping.level1.radius,
                            x: 0,
                            y: AppTokens.ElevationMapping.level1.dy)
                    .buttonStyle(PlainButtonStyle()) // 移除 SwiftUI 默认按钮样式，实现原子级样式控制
                    .padding(0) // contentPadding = PaddingValues(0.dp)

                    Text(action.label)
                        .font(AppTokens.TypographyTokens.label) // MaterialTheme.typography.labelMedium
                        .kerning(0.2) // letterSpacing
                        .foregroundColor(AppTokens.Colors.onBackground)
                }
                .frame(maxWidth: .infinity) // 确保每个 Column 均分空间，实现 SpaceAround
            }
        }
    }

    // 最近活动标题
    private func recentActivityHeader() -> some View {
        HStack {
            Text("Recent Activity")
                .font(AppTokens.TypographyTokens.title) // MaterialTheme.typography.titleMedium
                .foregroundColor(AppTokens.Colors.onBackground)
            Spacer()
            Button(action: {}) {
                Text("View All")
                    .font(AppTokens.TypographyTokens.label) // MaterialTheme.typography.labelMedium
                    .kerning(0.2) // letterSpacing
                    .foregroundColor(AppTokens.Colors.tertiary)
            }
            .buttonStyle(PlainButtonStyle())
        }
    }

    // 交易卡片
    private func transactionCard(transaction: Transaction) -> some View {
        HStack(spacing: AppTokens.Spacing.md) {
            AppTokens.Shapes.medium // Box(modifier = Modifier.size(48.dp).background(AppTokens.Colors.surfaceVariant, AppTokens.Shapes.medium))
                .fill(AppTokens.Colors.surfaceVariant)
                .frame(width: 48, height: 48)

            VStack(alignment: .leading) {
                Text(transaction.name)
                    .font(AppTokens.TypographyTokens.body) // MaterialTheme.typography.bodyMedium
                    .fontWeight(.semibold)
                    .foregroundColor(AppTokens.Colors.onSurface)
                Text(transaction.detail)
                    .font(AppTokens.TypographyTokens.label) // MaterialTheme.typography.labelMedium
                    .kerning(0.2) // letterSpacing
                    .foregroundColor(AppTokens.Colors.tertiary)
            }
            .frame(maxWidth: .infinity, alignment: .leading) // modifier = Modifier.weight(1f)

            Text(transaction.amount)
                .font(AppTokens.TypographyTokens.body) // MaterialTheme.typography.bodyMedium
                .fontWeight(.semibold)
                .foregroundColor(transaction.isCredit ? AppTokens.Colors.success : AppTokens.Colors.onSurface)
                .multilineTextAlignment(.trailing) // TextAlign.End
        }
        .padding(AppTokens.Spacing.md)
        .frame(maxWidth: .infinity) // fillMaxWidth()
        .background(AppTokens.Colors.surface) // containerColor
        .clipShape(AppTokens.Shapes.medium) // shape = AppTokens.Shapes.medium
        // border = BorderStroke(1.dp, AppTokens.Colors.outline)
        .overlay(
            AppTokens.Shapes.medium.stroke(AppTokens.Colors.outline, lineWidth: 1)
        )
    }

    // 底部导航栏
    private func bottomNavigationBar() -> some View {
        HStack(spacing: 0) {
            ForEach(navItems) { item in
                Button(action: {
                    selectedNavItem = item.label
                }) {
                    VStack(spacing: AppTokens.Spacing.xs) { // 模拟图标和文本的间距
                        Circle() // 图标占位符
                            .fill(selectedNavItem == item.label ? AppTokens.Colors.primary : AppTokens.Colors.outline)
                            .frame(width: 24, height: 24)
                        Text(item.label)
                            .font(AppTokens.TypographyTokens.label) // MaterialTheme.typography.labelMedium
                            .kerning(0.2) // letterSpacing
                            .foregroundColor(selectedNavItem == item.label ? AppTokens.Colors.primary : AppTokens.Colors.tertiary)
                    }
                    .frame(maxWidth: .infinity) // 每个导航项均分宽度
                    .padding(.vertical, AppTokens.Spacing.sm) // 模拟 NavigationBarItem 的垂直内边距
                    // indicatorColor 模拟，当选中时背景色变为 surfaceVariant
                    .background(selectedNavItem == item.label ? AppTokens.Colors.surfaceVariant : Color.clear)
                    .clipShape(RoundedRectangle(cornerRadius: AppTokens.Shapes.medium.cornerRadius)) // 裁剪指示器背景
                }
                .buttonStyle(PlainButtonStyle()) // 移除默认按钮样式
            }
        }
        .padding(.horizontal, AppTokens.Spacing.xs) // 整个导航栏的水平内边距
        .frame(height: 80) // 模拟 NavigationBar 的高度
        .background(AppTokens.Colors.surface) // containerColor
    }
}

// MARK: - App Entry Point (应用入口)
// @main 宏定义了应用的入口点，使其成为一个完整的可运行示例。
@main
struct SingleFileUIApp: App {
    var body: some Scene {
        WindowGroup {
            RootScreen()
        }
    }
}

// MARK: - Preview (预览)
struct RootScreen_Previews: PreviewProvider {
    static var previews: some View {
        RootScreen()
            .previewDisplayName("My Wallet UI")
    }
}